src/network/networkd-network.c

```diff
@@ -79,6 +79,47 @@ void network_config_section_free(NetworkConfigSection *cs) {
           free(cs);
 }
 
+/* Set defaults following RFC7844 */
+int network_apply_anonymize_if_set(Network *network) {
+        if (network->dhcp_anonymize) {
```

Comment:
i'd rewrite that this as:

```c
if (!network->dhcp_anonymize)
        return;
```

such an early exit has the benefit that we don't have to indent as much...